src/array.ts

```diff
@@ -355,6 +355,32 @@ export const replaceOrAppend = <T>(
   return [...list, newItem]
 }
 
+/**
+ * If the item matching the condition already exists
+ * in the list it will be removed. If it does not it
+ * will be added.
+ */
+export const toggle = <T>(
+  list: readonly T[],
+  newItem: T,
+  toValue?: null | ((a: T, idx: number) => number | string | symbol),
```

Comment:
I find the name `toValue` a little bit confusing because the value is not actually converted. However, I'm not sure which name would be better, maybe something like `compare`, `compareFn` or just `match` like in the other array functions?

Reply:
I add here that probably also `newItem` would be better if it was named simply `item`. The reason is that toggle can add a new item or remove an existing one, so I think `newItem` could be misleading

Reply:
For the `toValue` I agree it's a bit ambiguous, it's a little different than a match function though 🤔 I feel like `toKey` is a bit better but no matter it's a little odd so I added a comment. If ya'll feel strongly about match I'd be happy to change it -- I'm not that tied up on what it's called.

Agree on `item`, changed it 👍